client/session.go

```diff
@@ -1625,74 +1661,163 @@ func (s *session) FetchBlocksFromPeers(
 
 func (s *session) streamBlocksMetadataFromPeers(
 	namespace ident.ID,
-	shard uint32,
-	peers []peer,
+	shardID uint32,
+	peers peers,
 	start, end time.Time,
+	consistencyLevel ReadConsistencyLevel,
 	metadataCh chan<- blocksMetadata,
 	resultOpts result.Options,
 	progress *streamFromPeersMetrics,
 	version FetchBlocksMetadataEndpointVersion,
 ) error {
 	var (
 		wg       sync.WaitGroup
-		errLock  sync.Mutex
-		errLen   int
-		pending  int64
-		multiErr = xerrors.NewMultiError()
+		errLock  sync.RWMutex
+		errors   = make(map[int]error)
```

Comment:
nit: could you break the `errors` and `abortError` vars and associated methods into a helper struct, would help readability a bunch.

Reply:
+1

Reply:
Sure thing.